tests/contrib/falcon/test_distributed_tracing.py

```diff
@@ -0,0 +1,39 @@
+from ddtrace.propagation.http import HTTPPropagator
+from ddtrace.ext import errors as errx, http as httpx, AppTypes
+from falcon import testing
+from nose.tools import eq_, ok_
+from tests.test_tracer import get_dummy_tracer
+
+from .app import get_app
+
+
+class DistributedTracingTestCase(testing.TestCase):
+    """Executes tests using the manual instrumentation so a middleware
+    is explicitly added.
+    """
+
+    def setUp(self):
+        super(DistributedTracingTestCase, self).setUp()
+        self._service = 'falcon'
+        self.tracer = get_dummy_tracer()
+        self.api = get_app(tracer=self.tracer, distributed_tracing=True)
+
+    def test_has_parent_span(self):
+        headers = {}
+        root_tracer = get_dummy_tracer()
+        root_tracer.set_service_info('root', 'root', AppTypes.web)
+        with root_tracer.trace('root') as root:
+            propagator = HTTPPropagator()
+            propagator.inject(root.context, headers)
```

Comment:
can't you send just plain headers here? such as: https://github.com/DataDog/dd-trace-py/blob/d97830251b06b39d07dce512dd2ac1fb9d8e0704/tests/contrib/aiohttp/test_middleware.py#L212-L341

I would prefer to decouple in the test the `inject` and `extract` methods. If inject fails or does the wrong thing, we may wrongly assume it's working/not working on `extract`.

Reply:
I would replicate the same test also for disabled states. I think checking the tests in my previous link could help in improving covered cases (some of them may be merged because added in multiple iterations).

Reply:
Agree on hard-coding the headers.

Not sure what you mean about 'disabled states'. I added a test to ensure headers aren't turned into a parent when distributed tracing is disabled. 

Are there other useful cases I can get from the aiohttp tests you link? I looked at the Sampling tests but I'm not sure if they apply since none of my code looks at sampling.